### PR TITLE
Lab2 part 1

### PR DESCRIPTION
**Pull request by goltsmannn**

### CODE REVIEW

File: goltsmannn2/default.nix [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{ stdenv, cmake }:
+
+stdenv.mkDerivation rec {
+  pname = "lab2_part1";
+  version = "0.1.0";
+  src = "CMakeLists.txt";
```
Comment (RCMast3r) [inline goltsmannn2/default.nix:6]: this should just be the relative path to your source code, not just a single source:

https://blog.ielliott.io/nix-docs/mkDerivation.html

File: goltsmannn2/CMakeLists.txt
Comment (RCMast3r) [inline goltsmannn2/CMakeLists.txt:1]: you need to have an install step for your binary as the installed files are what nix actually copies to its nix store path

https://cmake.org/cmake/help/latest/command/install.html

File: goltsmannn2/default.nix (hunk 1 above)
Comment (RCMast3r) [reply goltsmannn2/default.nix:6]: more docs here: https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md

File: goltsmannn2/flake.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            system = "aarch64-darwin";
+            pkgs = import nixpkgs {
+                system = system;
+            };
+            hello_world_pkg = pkgs.callPackage ./default.nix {};
+      in
+      {
+        packages.aarch64-darwin = rec {
+            default = hello_world_pkg;
+        };
```
Comment (RCMast3r) [inline goltsmannn2/flake.nix:20]: this is a good flake! except for one thing which is we should prefer to support multiple system types per flake:

you can do this following this blog post's outline
https://ayats.org/blog/no-flake-utils

**Review by RCMast3r — CHANGES_REQUESTED** [review]

**Comment by goltsmannn** [comment]
Fixed everything

**Review by RCMast3r — APPROVED** [review]

**Pull request merged**